### PR TITLE
create image pushing job for kube-scheduler-simulator

### PR DESCRIPTION
**Pull request by sanposhiho**
Hello team. This PR creates image pushing job for kube-scheduler-simulator.

part of https://github.com/kubernetes-sigs/kube-scheduler-simulator/issues/19

staging repo was created in https://github.com/kubernetes/k8s.io/pull/2681

On the repo, cloudbuild.yaml will be created in https://github.com/kubernetes-sigs/kube-scheduler-simulator/pull/21

### CODE REVIEW

**Review by spiffxp — APPROVED** [review]
/approve
/lgtm

File: config/jobs/image-pushing/k8s-staging-sched-simulator.yaml [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+        containers:
+          - image: gcr.io/k8s-testimages/image-builder:v20210622-762366a
+            command:
+              - /run.sh
+            args:
+              # this is the project GCB will run in, which is the same as the GCR
+              # images are pushed to.
+              - --project=k8s-staging-sched-simulator
+              # This is the same as above, but with -gcb appended.
+              - --scratch-bucket=gs://k8s-staging-sched-simulator
```
Comment (spiffxp) [inline config/jobs/image-pushing/k8s-staging-sched-simulator.yaml:31]:
```suggestion
              - --scratch-bucket=gs://k8s-staging-sched-simulator-gcb
```

**Review by spiffxp — CHANGES_REQUESTED** [review]

File: config/jobs/image-pushing/k8s-staging-sched-simulator.yaml (hunk 1 above)
Comment (sanposhiho) [reply config/jobs/image-pushing/k8s-staging-sched-simulator.yaml]: Thanks :)

**Comment by sanposhiho** [comment]
@spiffxp thanks for reviewing, fixed it

**Review by spiffxp — APPROVED** [review]
/approve
/lgtm

**Comment by sanposhiho** [comment]
changed to use k8s-staging-test-infra/image-builder instead of k8s-testimages/image-builder.
Ref: https://github.com/kubernetes/test-infra/pull/23587

File: config/jobs/image-pushing/k8s-staging-sched-simulator.yaml [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+      # job makes sense on every branch (unless it's setting a `latest` tag it
+      # probably does).
+      branches:
+        - ^master$
+        # this is a regex for semver, from https://semver.org/#is-there-a-suggested-regular-expression-regex-to-check-a-semver-string
+        - ^v(0|[1-9]\d*)\.(0|[1-9]\d*)\.(0|[1-9]\d*)(?:-((?:0|[1-9]\d*|\d*[a-zA-Z-][0-9a-zA-Z-]*)(?:\.(?:0|[1-9]\d*|\d*[a-zA-Z-][0-9a-zA-Z-]*))*))?(?:\+([0-9a-zA-Z-]+(?:\.[0-9a-zA-Z-]+)*))?$
+      spec:
+        serviceAccountName: gcb-builder
+        containers:
+          - image: gcr.io/k8s-testimages/image-builder:v20210622-762366a
```
Comment (spiffxp) [inline config/jobs/image-pushing/k8s-staging-sched-simulator.yaml]:
```suggestion
          - image: gcr.io/k8s-staging-test-infra/image-builder:v20210913-fc7c4e84f6
```

**Comment by spiffxp** [comment]
Can you rebase to drop the merge commit

**Comment by sanposhiho** [comment]
> Can you rebase to drop the merge commit

👍  done

**Review by spiffxp — APPROVED** [review]
/approve
/lgtm

**Pull request merged**